CRM/Core/Payment/PayPalImpl.php

```diff
@@ -687,9 +687,20 @@ public function cancelSubscription(&$message = '', $params = array()) {
    * We don't need to handle paypal standard using this path as there has never been any historic support
    * for paypal standard to call civicrm/payment/ipn as a path.
    */
-  static public function handlePaymentNotification() {
-    $paypalIPN = new CRM_Core_Payment_PayPalProIPN($_REQUEST);
-    $paypalIPN->main();
+  public function handlePaymentNotification() {
+    $log = new CRM_Utils_SystemLogger();
```

Comment:
This should not be necessary - as long as this function is being reached through civicrm/payment/ipn/xx then this will happen before it gets to this point